shuup/admin/modules/products/mass_actions.py

```diff
@@ -28,7 +28,9 @@ def process(self, request, ids):
         query = Q(product__pk__in=ids)
         if isinstance(ids, string_types) and ids == "all":
             query = Q()
-        ShopProduct.objects.filter(query).update(visibility=ShopProductVisibility.ALWAYS_VISIBLE)
+        for shop_product in ShopProduct.objects.filter(query).iterator():
```

Comment:
I guess we have a bug here.. when the merchant selects ALL, it will basically set ALL products of ALL shops :scream: 
this is completely wrong.. fix this by adding the admin shop (from the request) in the basic query and make sure to test this